### PR TITLE
Remove `ln_gateway`

### PR DESCRIPTION
**Pull request by justinmoon**
Removes any types and implementations in support of `ln_gateway`
After PR #1717 , we can fully delete the old gateway, as proposed with this change

### CODE REVIEW

File: gateway/gateway/Cargo.toml [hunk 1]
```diff
@@ -1,15 +1,15 @@
 [package]
-name = "ln-gateway"
+name = "gateway"
 version = "0.1.0"
 authors = ["The Fedimint Developers"]
 edition = "2021"
-description = "ln-gateway is a core lightning plugin which allows a Lightning node operator to receive or pay Lightning invoices on behalf of fedimint users."
+description = "gateway is a core lightning plugin which allows a Lightning node operator to receive or pay Lightning invoices on behalf of fedimint users."
```
Comment (okjodom) [inline gateway/gateway/Cargo.toml:6]: we should update this description. perhaps this?

_gateway is an extension service that facilitates receiving or paying Lightning invoices on behalf of users in a connected Fedimint federation. Run this if you want to connect a federation to the rest of the bitcoin network, over lightning_

File: gateway/gateway/src/bin/cln_extension.rs [hunk 2]
```diff
@@ -43,7 +41,7 @@ pub struct ClnExtensionOpts {
 }
 
 // Note: Once this binary is stable, we should be able to remove current
```
Comment (okjodom) [inline gateway/gateway/src/bin/cln_extension.rs:43]: remove the note about removing `ln_gateway`?

File: scripts/gw-reload.sh [hunk 3]
```diff
@@ -3,9 +3,9 @@
 
 set -euo pipefail
 
-cargo build ${CARGO_PROFILE:+--profile ${CARGO_PROFILE}} --bin ln_gateway
+cargo build ${CARGO_PROFILE:+--profile ${CARGO_PROFILE}} --bin gateway
 
-$FM_LN1 plugin stop ln_gateway &> /dev/null || true
-$FM_LN1 -k plugin subcommand=start plugin=$FM_BIN_DIR/ln_gateway fedimint-cfg=$FM_CFG_DIR &> /dev/null
+$FM_LN1 plugin stop gateway &> /dev/null || true
+$FM_LN1 -k plugin subcommand=start plugin=$FM_BIN_DIR/gateway fedimint-cfg=$FM_CFG_DIR &> /dev/null
```
Comment (okjodom) [inline scripts/gw-reload.sh:9]: we should configure `gateway-cln-extension` as the plugin

File: docs/gateway.md [hunk 4]
```diff
@@ -42,8 +42,8 @@ A Lightning extension / plugin service that provides all the necessary Lightning
 
 > **Note**
 >
-> We are currenly transitioning from a monolith `ln_gateway` extension that runs all the gateway business logic inside of a cln extension.
-> `ln_gateway` is equivalent to `Gatewayd` and `gateway-cln-extension` packaged into the same binary to run inside a CLN node as a plugin.
+> We are currenly transitioning from a monolith `gateway` extension that runs all the gateway business logic inside of a cln extension.
```
Comment (okjodom) [inline docs/gateway.md:45]: let's remove this comment

File: fedimint-testing/src/ln/fixtures.rs [hunk 5]
```diff
@@ -9,14 +9,14 @@ use bitcoin::{secp256k1, KeyPair};
...
-use lightning::ln::PaymentSecret;
-use lightning_invoice::{Currency, Invoice, InvoiceBuilder, SignedRawInvoice, DEFAULT_EXPIRY_TIME};
-use ln_gateway::gatewaylnrpc::{
+use gateway::gatewaylnrpc::{
     self, CompleteHtlcsRequest, CompleteHtlcsResponse, GetPubKeyResponse, GetRouteHintsResponse,
     PayInvoiceRequest, PayInvoiceResponse, SubscribeInterceptHtlcsRequest,
 };
-use ln_gateway::ln::{LightningError, LnRpc};
-use ln_gateway::lnrpc_client::{HtlcStream, ILnRpcClient};
+use gateway::ln::{LightningError, LnRpc};
```
Comment (okjodom) [inline fedimint-testing/src/ln/fixtures.rs:16]: we should be able to delete `gateway/ln.rs`

File: docs/architecture.md [hunk 6]
```diff
@@ -23,8 +23,8 @@ The [user client](#User-Client):
 * `client/client-lib` - provides a library for sending transactions to the federation
 * `client/cli` - cli wrapper around the client library
 
-The [LN gateway](#LN-Gateway):
-* `ln-gateway` - allows a Lightning node operator to receive or pay Lightning invoices on behalf of users
+The [LN gateway](#gateway):
+* `gateway` - allows a Lightning node operator to receive or pay Lightning invoices on behalf of users
```
Comment (justinmoon) [inline docs/architecture.md:27]: wrong

File: docs/networking.md [hunk 7]
```diff
@@ -4,7 +4,7 @@ Default ports:
 
 * fedimintd p2p: 8173
 * fedimint API: 8174
-* ln-gateway API: 8175
+* gateway API: 8175
```
Comment (justinmoon) [inline docs/networking.md:7]: gatewayd

File: flake.nix [hunk 8]
```diff
@@ -767,7 +767,7 @@
                     rpc-file-mode=0660
                     log-timestamps=false
 
-                    plugin=${ln-gateway-pkgs}/bin/ln_gateway
+                    plugin=${gateway-pkgs}/bin/gateway
```
Comment (justinmoon) [inline flake.nix:770]: wrong

File: flake.nix [hunk 9]
```diff
@@ -779,11 +779,11 @@
...
                 in
                 pkgs.dockerTools.buildLayeredImage {
-                  name = "ln-gateway-clightning";
-                  contents = [ ln-gateway-pkgs clightning-dev pkgs.bash pkgs.coreutils ];
+                  name = "gateway-clightning";
+                  contents = [ gateway-pkgs clightning-dev pkgs.bash pkgs.coreutils ];
                   config = {
                     Cmd = [
-                      "${ln-gateway-pkgs}/bin/ln_gateway"
+                      "${gateway-pkgs}/bin/gateway"
```
Comment (justinmoon) [inline flake.nix:786]: wrong

File: gateway/gateway/Cargo.toml (hunk 1 above)
Comment (justinmoon) [reply gateway/gateway/Cargo.toml]: let's do that later

**Comment by justinmoon** [comment]
ok that's small

**Review by dpc — APPROVED** [review]

**Review by okjodom — DISMISSED** [review]

**Review by mxz42 — APPROVED** [review]

**Comment by mxz42** [comment]
might not be directly related here but running `just test-real` fails because my limit for open files is too small. I saw your `just check-ulimit`  @dpc but my limit was already set to 1024. setting it to 2048 works; perhaps we want to adjust it if we see it multiple times. 

also after tests I have a process running: `gateway-cln-ext` that is not killed. 

relevant output: 
```sh
$ ss -tlpn
LISTEN               0                    128                                             0.0.0.0:8177                                     0.0.0.0:*                   users:(("gateway-cln-ext",pid=837280,fd=9))
```

I'm not exactly sure where this is coming from. do we set the program name in rust? I can't find a file named `gateway-cln-ext`
```sh
 (pr/1718 %) | fd -HI gateway-cln
target/debug/.fingerprint/ln-gateway-7297e04178f8dea5/bin-gateway-cln-extension
target/debug/.fingerprint/ln-gateway-7297e04178f8dea5/bin-gateway-cln-extension.json
target/debug/.fingerprint/ln-gateway-854dd9a992bd47f8/test-bin-gateway-cln-extension
target/debug/.fingerprint/ln-gateway-854dd9a992bd47f8/test-bin-gateway-cln-extension.json
target/debug/.fingerprint/ln-gateway-b02b706c0674c161/bin-gateway-cln-extension
target/debug/.fingerprint/ln-gateway-b02b706c0674c161/bin-gateway-cln-extension.json
target/debug/.fingerprint/ln-gateway-b02b706c0674c161/dep-bin-gateway-cln-extension
target/debug/gateway-cln-extension
target/debug/gateway-cln-extension.d
```

**Comment by justinmoon** [comment]
CI failing here https://github.com/fedimint/fedimint/actions/runs/4257087817/jobs/7410793182#step:10:874

Can't reproduce locally

File: gateway/ln-gateway/src/lib.rs [hunk 10]
```diff
@@ -203,7 +204,7 @@ impl Gateway {
         payload: ConnectFedPayload,
         route_hints: Vec<RouteHint>,
     ) -> Result<()> {
-        let connect: WsClientConnectInfo = serde_json::from_str(&payload.connect).map_err(|e| {
+        let connect = WsClientConnectInfo::from_str(&payload.connect).map_err(|e| {
```
Comment (elsirion) [inline gateway/ln-gateway/src/lib.rs:207]: Ah right, fixup for #1660

File: gateway/ln-gateway/src/lib.rs [hunk 11]
```diff
@@ -172,7 +170,10 @@ impl Gateway {
             .await
             .get(&federation_id.to_string())
             .cloned()
-            .ok_or(GatewayError::UnknownFederation)
+            .ok_or(GatewayError::Other(anyhow::anyhow!(
+                "No federation with id {}",
+                federation_id.to_string()
```
Comment (elsirion) [inline gateway/ln-gateway/src/lib.rs:176]: `.to_string()` isn't necessary here (assuming it is available because the type implements `Display`)

**Review by elsirion — APPROVED** [review]
High level looks good, someone else should take a closer look.

File: gateway/ln-gateway/src/lib.rs (hunk 10 above)
Comment (okjodom) [reply gateway/ln-gateway/src/lib.rs:207]: Yeap!

After fix, I can't replicate [CI failure reported by Justin](https://github.com/fedimint/fedimint/pull/1718#issuecomment-1442766981)

File: gateway/ln-gateway/src/lib.rs (hunk 11 above)
Comment (mxz42) [reply gateway/ln-gateway/src/lib.rs:176]: Would be a good idea to implement `Display` for `FederationId` as it's not implemented yet

**Review by mxz42 — APPROVED** [review]
I think this is good to go. 

- the fix that `gateway-cln-extension` shuts down when clightning shuts down will be a separate PR. So far, if you run the tests (`just test-real`) the process `gateway-cln-extension` does not terminate. we can kill it with `pkill -f gateway-cln-extension` until the fix is available.
- remove the `.to_string()` elsirion mentioned when `Display` is implemented for `FederationId`

**Pull request merged**